File: Data Structures & Algorithms/merge-intervals/submission-0.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        intervals.sort(key=lambda a: a[0])
        start = intervals[0][0]
        end = intervals[0][1]
        out = list()
        for interval in intervals[1:]:
            if end < interval[0]:
                out.append([start,end])
                start = interval[0]
                end = interval[1]
            else:
                end = max(interval[1], end)
        out.append([start,end])
        return out

File: Data Structures & Algorithms/longest-increasing-subsequence/submission-0.java
class Solution {
    public int lengthOfLIS(int[] nums) {
        int l = nums.length;
        int[] longests = new int[l];
        int out = 0;
        for (int i = l - 1; i >= 0; i--) {
            int max = 1;
            for (int j = i + 1; j < l; j++) {
                if (nums[i] < nums[j]) {
                    if (longests[j] + 1 > max) {
                        max = longests[j] + 1;
                    }
                }
            }
            longests[i] = max;
            if (max > out) {
                out = max;
            }
        }
        return out;
    }
}
